configure.ac

```diff
@@ -59,6 +59,15 @@ AS_CASE($ax_cv_cxx_compiler_vendor,
 AX_CXX_COMPILE_STDCXX([11], [noext], [mandatory])
 
 X_AC_ENABLE_SANITIZER
+if test "x$san_enabled" != "xno" ; then
+  AC_DEFINE([DEEPBIND], [0],
+            [deepbind is unsupported with asan, musl and so-forth])
```

Comment:
I might suggest a different name for the `DEEPBIND` config.h macro. Perhaps `FLUX_DEEPBIND` to give a hint in the code that we're using a locally provided define.

Reply:
Good point @grondo.  It also made me notice I had forgotten to add the check for whether RTLD_DEEPBIND even exists, so it's renamed and the check has been added as well.